typescript/infra/src/utils/rpcUrls.ts

```diff
@@ -1,46 +1,57 @@
+import input from '@inquirer/input';
+import { Separator, checkbox, confirm } from '@inquirer/prompts';
+import select from '@inquirer/select';
 import { ethers } from 'ethers';
 
-import { timeout } from '@hyperlane-xyz/utils';
+import { ChainName } from '@hyperlane-xyz/sdk';
+import { ProtocolType, timeout } from '@hyperlane-xyz/utils';
 
+import { Contexts } from '../../config/contexts.js';
+import { getChain } from '../../config/registry.js';
+import { getEnvironmentConfig } from '../../scripts/core-utils.js';
 import {
+  RelayerHelmManager,
+  ScraperHelmManager,
+  ValidatorHelmManager,
   getSecretRpcEndpoints,
   getSecretRpcEndpointsLatestVersionName,
   secretRpcEndpointsExist,
   setSecretRpcEndpoints,
 } from '../agents/index.js';
+import { DeployEnvironment } from '../config/environment.js';
+import { KeyFunderHelmManager } from '../funding/key-funder.js';
+import { KathyHelmManager } from '../helloworld/kathy.js';
 
 import { disableGCPSecretVersion } from './gcloud.js';
-import { isEthereumProtocolChain } from './utils.js';
+import { HelmManager } from './helm.js';
+import { K8sResourceType, refreshK8sResources } from './k8s.js';
 
-export async function testProviders(rpcUrlsArray: string[]): Promise<boolean> {
-  let providersSucceeded = true;
-  for (const url of rpcUrlsArray) {
-    const provider = new ethers.providers.StaticJsonRpcProvider(url);
-    try {
-      const blockNumber = await timeout(provider.getBlockNumber(), 5000);
-      console.log(`Valid provider for ${url} with block number ${blockNumber}`);
-    } catch (e) {
-      console.error(`Provider failed: ${url}`);
-      providersSucceeded = false;
-    }
-  }
-
-  return providersSucceeded;
-}
-
-export async function setAndVerifyRpcUrls(
+/**
+ * Set the RPC URLs for the given chain in the given environment interactively.
+ * Includes an interactive experience for selecting new RPC URLs, confirming the change,
+ * updating the secret, and refreshing dependent k8s resources.
+ * @param environment The environment to set the RPC URLs in
+ * @param chain The chain to set the RPC URLs for
+ */
+export async function setRpcUrlsInteractive(
   environment: string,
   chain: string,
-  rpcUrlsArray: string[],
 ): Promise<void> {
-  const secretPayload = JSON.stringify(rpcUrlsArray);
-
   try {
-    await displayCurrentSecrets(environment, chain);
-    await confirmSetSecrets(environment, chain, secretPayload);
-    await testProvidersIfNeeded(chain, rpcUrlsArray);
+    const currentSecrets = await getAndDisplayCurrentSecrets(
+      environment,
+      chain,
+    );
+    const newRpcUrls = await inputRpcUrls(chain, currentSecrets);
+    console.log(`Selected RPC URLs: ${formatRpcUrls(newRpcUrls)}\n`);
+
+    const secretPayload = JSON.stringify(newRpcUrls);
+    await confirmSetSecretsInteractive(environment, chain, secretPayload);
     await updateSecretAndDisablePrevious(environment, chain, secretPayload);
+    await refreshDependentK8sResourcesInteractive(
```

Comment:
nit: would be nice to have:
- `--no-rotate` or `--set-only` to skip the refresh step
- `--rotate-only` to skip the set/update secrets steps

Reply:
you can skip it interactively with the Y/n prompt, are you saying a CLI flag would be preferable?

Reply:
yeah just as an additional option to save some reading/clicks, but not required